### PR TITLE
feat: add warning message for the info command

### PR DESCRIPTION
**Pull request by GreGosPhaTos**
More details here: https://github.com/nestjs/nest-cli/issues/1929

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Other... Please describe:

## What is the current behavior?
There is no warnin message if deveopers use different version of the nest-js package in a projet

Issue Number: https://github.com/nestjs/nest-cli/issues/1929

## What is the new behavior?

Developer now gets a warning message folowing the suggestion => https://github.com/nestjs/nest-cli/issues/1929

Capture d’écran, le 2023-11-22 à 09 38 28

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by micalevisk** [comment]
note that this rule doesn't apply to all packages under `@nestjs/*`. For example, the latest version of `@nestjs/swagger` is v7, not v10.

thus we need a whitelist of those that will be checked against. I think you can start wit this:
- `@nestjs/core`
- `@nestjs/common`
- `@nestjs/schematics`
- `@nestjs/platform-express`
- `@nestjs/platform-fastify`

I'm not sure about the others. I guess Kamil should tell us that

**Comment by GreGosPhaTos** [comment]
@micalevisk added the white list directly in the method https://github.com/nestjs/nest-cli/pull/2396/files#diff-088f2456e09a57e3cc418bccd4ab7a7d3968fb4fe3cd8e24a358f76ba6ad1473R142-R151

File: actions/info.action.ts [hunk 1]
```diff
@@ -21,6 +21,11 @@ interface PackageJsonDependencies {
 interface NestDependency {
   name: string;
   value: string;
+  fullName: string;
```
Comment (micalevisk) [inline actions/info.action.ts:24]: can we rename `fullName` to `packageName`?

File: actions/info.action.ts [hunk 2]
```diff
@@ -104,9 +109,58 @@ export class InfoAction extends AbstractAction {
...
+      })
+    }
+  }
+
+  buildNestVersionsWarningMessage(
+    nestDependencies: NestDependency[],
+  ): NestDependencyWarnings {
+    const unsortedWarnings: NestDependencyWarnings = nestDependencies.reduce<NestDependencyWarnings>(
+      (acc, { name, fullName, value }) => {
+        const [major, minor] = value.split('.').map(parseFloat);
```
[the remark this answers is not in the document]
Comment (kamilmysliwiec) [reply actions/info.action.ts:159]: we might need to add some error handling here

[the remark this answers is not in the document]
Comment (GreGosPhaTos) [reply actions/info.action.ts]: I added [this](https://github.com/nestjs/nest-cli/pull/2396/files#diff-088f2456e09a57e3cc418bccd4ab7a7d3968fb4fe3cd8e24a358f76ba6ad1473R159) which filter non-digits chars out of the string. 
I also added more test cases.
The main "risk" would be in the sort callback, and I added some guards as well see => https://github.com/nestjs/nest-cli/pull/2396/files#diff-088f2456e09a57e3cc418bccd4ab7a7d3968fb4fe3cd8e24a358f76ba6ad1473R177-R190 

Eventually I can throw an error and catch it in the [displayWarningMessage](https://github.com/nestjs/nest-cli/pull/2396/files#diff-088f2456e09a57e3cc418bccd4ab7a7d3968fb4fe3cd8e24a358f76ba6ad1473R120) method to display a generic error message, but I think I covered all the cases with the last changes. What's you toughts @kamilmysliwiec  ?

File: actions/info.action.ts (hunk 1 above)
Comment (GreGosPhaTos) [reply actions/info.action.ts]: done

**Review by micalevisk — APPROVED** [review]

File: actions/info.action.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kamilmysliwiec) [reply actions/info.action.ts]: > Eventually I can throw an error and catch it in the [displayWarningMessage](https://github.com/nestjs/nest-cli/pull/2396/files#diff-088f2456e09a57e3cc418bccd4ab7a7d3968fb4fe3cd8e24a358f76ba6ad1473R120) method to display a generic error message, but I think I covered all the cases with the last changes. What's you toughts @kamilmysliwiec ?

We could add it _just in case_

[the remark this answers is not in the document]
Comment (GreGosPhaTos) [reply actions/info.action.ts]: Done => https://github.com/nestjs/nest-cli/pull/2396/commits/d29182ffda12f708764b6ca91fbfb1c7c755be1d

**Comment by GreGosPhaTos** [comment]
Just wondering is this PR needs some changes ? Otherwise should it be merged soon ? 

@micalevisk @kamilmysliwiec

**Comment by micalevisk** [comment]
Looks fine to me so far

I guess it will be merged after Kamil get some time to review this too, don't worry.

**Comment by micalevisk** [comment]
@GreGosPhaTos btw could you please open a PR at https://github.com/nestjs/docs.nestjs.com to document this new behavior of the `info` command? I guess new subsection under https://docs.nestjs.com/cli/usages#nest-info would be good

**Comment by GreGosPhaTos** [comment]
> @GreGosPhaTos btw could you please open a PR at https://github.com/nestjs/docs.nestjs.com to document this new behavior of the `info` command? I guess new subsection under https://docs.nestjs.com/cli/usages#nest-info would be good

@micalevisk Yes ok => here it is https://github.com/nestjs/docs.nestjs.com/pull/2923

**Review by rubiin — APPROVED** [review]

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**